usaspending_api/common/helpers/generic_helper.py

```diff
@@ -59,9 +60,15 @@ def generate_fiscal_month(date):
     """ Generate fiscal period based on the date provided """
     validate_date(date)
 
-    if date.month in [10, 11, 12, "10", "11", "12"]:
-        return int(date.month) - 9
-    return int(date.month) + 3
+    if date.month in [10, 11, 12]:
```

Comment:
_If this recommendation cascades into problems.... ignore._


A new "base class" for obtaining the fiscal date which is used for the other functions:

```python
def convert_date_to_fiscal_cal(date: datetime) -> FiscalDate:
    validate_date(date)
    return FiscalDate(date.year, date.month, date.day)

def generate_fiscal_quarter(date: datetime) -> int:
    return convert_date_to_fiscal_cal(date).quarter

# repeat for year and month....
```

It might encourage devs to migrate to using `convert_date_to_fiscal_cal()` instead of calling several of the other functions.

Could even use datetime.now(timezone.utc) as the default date since that is a common value passed into the functions

Reply:
Like the idea, but I feel that this will result in more changes then what I would like to make right now if not needed.